DEBUG_android.md

```diff
@@ -0,0 +1,150 @@
+<!-- markdownlint-disable MD033 -->
+<!-- markdownlint-disable MD041 -->
+<p align="left"><img src="https://vulkan.lunarg.com/img/NewLunarGLogoBlack.png" alt="LunarG" width=263 height=113 /></p>
+
+[![Creative Commons][1]][2]
+
+[1]: https://i.creativecommons.org/l/by-nd/4.0/88x31.png "Creative Commons License"
+[2]: https://creativecommons.org/licenses/by-nd/4.0/
+
+Copyright &copy; 2018-2025 LunarG, Inc.
+
+# GFXReconstruct Capture layer debugging - Android
+
+***This document describes how to debug the GFXReconstruct capture layer
+on Android using lldb-server.***
+
+## Index
+
+1. [Problem](#problem)
+2. [Setting up lldb-server](#setting-up-lldb-server)
+3. [Client setup](#client-setup)
+   1. [VSCode](#vscode)
+   2. [CLI](#cli)
+4. [Waiting for debugger](#waiting-for-debugger)
+
+## Problem
+
+Android apps under development can be debugged easily through Android Studio. However, the GFXR capture layer is
+a shared object file that is loaded by the, typically non-debug, application. This document illustrates a 
+method by which the `lldb-server` binary included with the Android NDK can be used for debugging
+a .so built in debug mode with any Android app.
+
+## Setting up lldb-server
+
+The first step is to locate and copy the lldb-server binary to a location on the Android device,
+making sure to give it execute permissions as well.
+```
+adb root
+adb push $(find $ANDROID_NDK_HOME -path "*/aarch64/lldb-server") /data/local/tmp
+adb shell chmod +x /data/local/tmp/lldb-server
+```
+
+Then, run the following command to start the server listening on port 9999
+```
+adb shell /data/local/tmp/lldb-server platform --server --listen localhost:9999
+```
+
+lldb-server is now ready to accept connections from a client debugger.
+
+## Client setup
+
+On the client side, we need to launch `lldb` with options to do the following:
+  1. Disable breaking on signals we don't care about (e.g. SIGSEGV)
+  2. Load the capture layer's debug symbols
+  3. Attach to the remote debugger on the phone
+
+You need two pieces of information before continuing:
+  1. Your device's ID
+  2. The path to the directory containing the compiled Android layer e.g. `android/layer/build/intermediates/cxx/Debug/3m4on72q/obj/arm64-v8a/`
+
+You can obtain your device's id with the `adb devices` command:
+```
+> adb devices
+List of devices attached
+44021FDJH00229  device
+```
+
+We document two methods for attaching a client debugger to lldb-server: VScode, and CLI
+
+### VScode
+To enable VScode to be the client debugger, you need to set up your
```

Comment:
@mikes-lunarg have you used this method?  I've never used it and can't comment on the completion of this documented process.

Reply:
Yep I have tested this method. The roundabout way of getting the target PID is a little goofy but it works!